platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/StandardThreadFactoryBuilder.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class StandardThreadFactoryBuilder implements ThreadFactory {
+
+    public static ThreadFactory newThreadFactory(String threadNamePrefix) {
+        return newThreadFactory(threadNamePrefix, true, Thread.NORM_PRIORITY);
+    }
+
+    public static ThreadFactory newThreadFactory(String threadNamePrefix, boolean isDaemon, int priority) {
+        if (threadNamePrefix == null || threadNamePrefix.isEmpty()) {
+            threadNamePrefix = "thread";
+        }
+
+        return build(threadNamePrefix, isDaemon, priority);
+    }
+
+    private static ThreadFactory build(String threadName, boolean isDaemon, int priority) {
+        final AtomicLong count = new AtomicLong(0);
```

Comment:
_I'm re-adding this comment because it was lost when I posted my review._

❗️ It's early on a Monday morning so forgive me if I'm missing something...but this is going to create a new AtomicLong on the stack every time you call build; you'll always return a thread with a suffix of 0. Move it out of the method and make it a static final const.

Reply:
I wrote this over the last few DevDays, so I wish I could say I knew what I was thinking. Possibly an artifact of a minor refactor?

Reply:
Looking at this again, I remember the idea was that all new thread pools/factories would have a count independent of each other. I suppose this doesn't provide much value in practice though, especially considering your point about the stack cost.